### ISSUE TITLE
`coordinator finalize` command fails when there is no contribution

### ISSUE THREAD

**Issue by gurrpi** [quadratic-funding/mpc-phase2-suite#190]
```
$ yarn coordinator finalize

Greetings, @gurrpi 👋

⚠ The computation of the final contribution could take the bulk of your computational resources and memory based on the size of the circuit 🔥

✔ Select a ceremony › c3
✖ Something went wrong: FirebaseError: internal
```

## Cause

Where those error happened? 
* CLI side :https://github.com/quadratic-funding/mpc-phase2-suite/blob/c48df8d363b8e4fae26aced53b5174514b1bd773/phase2cli/src/commands/finalize.ts#L71
* Cloud Functions side: https://github.com/quadratic-funding/mpc-phase2-suite/blob/aec55044395462ac8dba27e5384d1c9d5f354cd5/firebase/src/finalize.ts#L58

Cloud Functions side log:
```
2022-11-12 22:00:21.008 CET
checkAndPrepareCoordinatorForFinalization3mcyflx06hnk Function execution started

2022-11-12 22:00:23.625 CET
checkAndPrepareCoordinatorForFinalization3mcyflx06hnk Callable request verification passed

2022-11-12 22:00:24.774 CET
checkAndPrepareCoordinatorForFinalization3mcyflx06hnk [ERROR] Data not found

2022-11-12 22:00:24.783 CET
checkAndPrepareCoordinatorForFinalization3mcyflx06hnk Function execution took 3774 ms. Finished with status: crash
```

**Comment by gurrpi** [comment]
A question arises in my mind:
What is supposed to do when there was no contribution?

**Comment by 0xjei** [comment]
I see. To answer your question, IMHO we shouldn't make the ceremony available for finalization. We could use this opportunity to better write down the statuses of ceremonies and participants by going on to make explicit the conditions that must exist to be in a particular status.

**Comment by gurrpi** [comment]
> I see. To answer your question, IMHO we shouldn't make the ceremony available for finalization. We could use this opportunity to better write down the statuses of ceremonies and participants by going on to make explicit the conditions that must exist to be in a particular status.
> 
> BTW, this can be easily solved by just adding a condition to check if at least one valid contribution has been produced for the ceremony before finalizing.

I agree. Maybe it should automatically removed from storage side after grace period and notify coordinator it. Otherwise storage cost would billed constantly.

**Comment by 0xjei** [comment]
> I agree. Maybe it should automatically removed from storage side after grace period and notify coordinator it. Otherwise storage cost would billed constantly.

I definitely agree with you, very cool proposal! Great insight :+1: Feel free to open a new issue to keep better track.
